### PR TITLE
[NEW] Modal

### LINKED ISSUE #8775: New Modal Component
### Description:

Create a new popup component to replace sweet alert.

### PR DESCRIPTION
**Pull request by karlprieb**
@RocketChat/core 

Closes #8775

Our new modal system :) This PR implement that and remove sweetalert :)

### CODE REVIEW

File: packages/rocketchat-livechat/app/client/lib/commands.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* globals LivechatVideoCall, Livechat */
+/* globals LivechatVideoCall, Livechat, swal */
```
Comment (geekgonecrazy) [inline packages/rocketchat-livechat/app/client/lib/commands.js]: is it still needed to add swal to globals when actually removing it?

**Comment by geekgonecrazy** [comment]
can we add some key press love up in here? 
```
onkeydown = function (event) { 
  if (event.which == 13) {
    modal.confirm(true);
  }

  if (event.which == 27) {
    modal.close();
  }
}
```

File: packages/rocketchat-livechat/app/client/lib/commands.js (hunk 1 above)
Comment (karlprieb) [reply packages/rocketchat-livechat/app/client/lib/commands.js]: Livechat widget still uses sweetAlert

Comment (geekgonecrazy) [reply packages/rocketchat-livechat/app/client/lib/commands.js]: :+1: makes sense I just saw the adding in a bunch of files so was checking :)

**Comment by karlprieb** [comment]
@graywolf336 @geekgonecrazy I added ESC and Enter events now.

File: packages/rocketchat-livechat/app/client/lib/commands.js (hunk 1 above)
Comment (karlprieb) [reply packages/rocketchat-livechat/app/client/lib/commands.js]: 👍

File: packages/rocketchat-theme/client/imports/components/modal.css [hunk 2]
```diff
@@ -56,14 +53,50 @@
...
+		align-items: center;
 
 		padding: 16px;
 
 		animation: dropdown-show 0.1s cubic-bezier(0.45, 0.05, 0.55, 0.95);
+	}
+
+	&__content-icon {
+		font-size: 6rem;
+    margin: 1.2rem;
```
Comment (sampaiodiego) [inline packages/rocketchat-theme/client/imports/components/modal.css]: change for tabs

File: packages/rocketchat-theme/client/imports/components/modal.css [hunk 3]
```diff
@@ -56,14 +53,50 @@
...
+		}
+
+		&--modal-error {
+			color: var(--rc-color-error-light);
+		}
+	}
 
-		white-space: nowrap;
+	&__content-text {
+    line-height: 1.5;
```
Comment (sampaiodiego) [inline packages/rocketchat-theme/client/imports/components/modal.css]: change for tabs

File: packages/rocketchat-theme/client/imports/components/modal.css [hunk 4]
```diff
@@ -56,14 +53,50 @@
...
+    text-align: center;
+		font-size: 1rem;
+		word-break: break-word;
+	}
 
-		background-color: var(--modal-background);
+	&__content-error {
+		display: none;
+		width: 100%;
+    text-align: center;
```
Comment (sampaiodiego) [inline packages/rocketchat-theme/client/imports/components/modal.css]: change for tabs

File: packages/rocketchat-ui/client/views/app/popover.js [hunk 5]
```diff
@@ -237,23 +237,19 @@ Template.popover.events({
...
 								html: false
 							});
 						} else {
-							swal.close();
+							modal.close();
 							if (['channel', 'group', 'direct'].includes(FlowRouter.getRouteName()) && (Session.get('openedRoom') === rid)) {
 								FlowRouter.go('home');
 							}
-
-							RoomManager.close(rid);
```
Comment (sampaiodiego) [inline packages/rocketchat-ui/client/views/app/popover.js]: add this back pls

**Comment by geekgonecrazy** [comment]
@karlprieb you rule!  Balance and Order has been restored to the universe.  :smile:

File: .eslintrc [hunk 6]
```diff
@@ -122,6 +122,7 @@
 		"Logger"                      : false,
 		"Match"                       : false,
 		"Meteor"                      : false,
+		"modal"												: false,
```
Comment (graywolf336) [inline .eslintrc]: Should this be aligned with the others?

**Pull request merged**